tests/MongoDB.Driver.Core.Tests/Core/Operations/CountDocumentsOperationTests.cs

```diff
@@ -0,0 +1,609 @@
+            {
+                MaxTime = milliseconds == null ? (TimeSpan?)null : TimeSpan.FromMilliseconds(milliseconds.Value)
+            };
+
+            // TODO: use failpoints to force a timeout?
+            var result = ExecuteOperation(subject, async);
+
+            result.Should().Be(2);
+        }
+
+        [SkippableTheory]
+        [ParameterAttributeData]
+        public void Execute_should_return_expected_result_when_ReadConcern_is_set(
+            [Values(null, ReadConcernLevel.Local)]
+            ReadConcernLevel? level,
+            [Values(false, true)]
+            bool async)
+        {
+            RequireServer.Check().Supports(Feature.ReadConcern);
+            EnsureTestData();
+            var readConcern = level == null ? ReadConcern.Default : new ReadConcern(level.Value);
+            var subject = new CountDocumentsOperation(_collectionNamespace, _messageEncoderSettings)
+            {
+                ReadConcern = readConcern
+            };
+
+            var result = ExecuteOperation(subject, async);
+
+            result.Should().Be(2);
+        }
+
+        [SkippableTheory]
+        [ParameterAttributeData]
+        public void Execute_should_throw_when_ReadConcern_is_set_but_not_supported(
+            [Values(false, true)]
+            bool async)
+        {
+            RequireServer.Check().DoesNotSupport(Feature.ReadConcern);
+            var subject = new CountDocumentsOperation(_collectionNamespace, _messageEncoderSettings)
+            {
+                ReadConcern = new ReadConcern(ReadConcernLevel.Local)
+            };
+
+            var exception = Record.Exception(() => ExecuteOperation(subject, async));
+
+            exception.Should().BeOfType<MongoClientException>();
+        }
+
+        [SkippableTheory]
+        [ParameterAttributeData]
+        public void Execute_should_return_expected_result_when_Skip_is_set(
+            [Values(null, 1L, 2L)]
+            long? skip,
+            [Values(false, true)]
+            bool async)
+        {
+            RequireServer.Check();
+            EnsureTestData();
+            var subject = new CountDocumentsOperation(_collectionNamespace, _messageEncoderSettings)
+            {
+                Skip = skip
+            };
+
+            var result = ExecuteOperation(subject, async);
+
+            result.Should().Be(2 - (skip ?? 0));
+        }
+
+        [SkippableTheory]
+        [ParameterAttributeData]
+        public void Execute_should_send_session_id_when_supported(
+            [Values(false, true)] bool async)
+        {
+            RequireServer.Check();
+            EnsureTestData();
+            var subject = new CountDocumentsOperation(_collectionNamespace, _messageEncoderSettings);
+
+            VerifySessionIdWasSentWhenSupported(subject, "aggregate", async);
+        }
+
+        [Fact]
+        public void CreateOperation_should_return_expected_result()
```

Comment:
This is where this file really starts deviating from CountOperationTests because we're testing completely different private methods.